### DISCUSSION TITLE
Cache question: create entity followed by querying that same entity

### DISCUSSION THREAD

**Discussion by smontlouis** [urql-graphql/urql#2415] [Q&A] [answered]
In the application the user can write notes. When clicking on the "create a note" action, a `createUserNote` mutation is sent and then redirects to `/notes/:id` which is responsible for querying the note.

I thought that if I returned the exact same data between the mutation and the query, urql would be able to use the cache to retrieve the note without fetching again.

Am I missing something?
If not, what would be the other possibilities, updating the cache manually? If so, how?

I'm using schema awareness and normalized cache

```gql
  fragment UserNoteData on UserNoteEntity {
    __typename
    id
    attributes {
      content
      updatedAt
      path
    }
  }


  query userNote($noteId: ID!) {
    userNote(id: $noteId) {
      data {
        ...UserNoteData
      }
    }
  }

  mutation createUserNote($data: UserNoteInput!) {
    createUserNote(data: $data) {
      data {
        ...UserNoteData
      }
    }
  }
`
```

**Comment by JoviDeCroock** [comment, marked as answer]
This is a two-fold answer, on one side when you create an entity we don't know how the entry points connect to that entity i.e. query.Todo(id: 1) or where in a list it lives. Secondly our cache does garbage collection so every disconnected node, no entry-points leading to it, will be deleted.

Therefor we advise using a custom updater that leverages updateQuery or cache.link to tell the cache how you will be reacghing this new entity.

**Reply by smontlouis** [reply]
Thank you for your answer.
I thought schema introspection would have been able to deal with it.

I was able to get a result by using local resolver, like the following :
```
userNote: (parent, args, cache, info) => ({
      __typename: 'UserNoteEntityResponse',
      data: {
        __typename: 'UserNoteEntity',
        id: args.id,
      },
    }),
```

Is this the right way to do it ? Does it mean that for each query I'll need to define local resolvers ?

**Reply by JoviDeCroock** [reply]
There's two ways you can hint with a resolver which will also help with the list --> detail resolution or you write an updater.

**Reply by smontlouis** [reply]
Yes I'm already using `updateQuery` to updates userNotes list.

I want to make sure I understand: 

- Write an updater
Meaning in the updates: { Mutation: { createUserNote: () => { ... I should be able to update the links in the cache ?
```
cache.link({ __typename: 'Query' }, 'userNote',{
      __typename: 'UserNoteEntityResponse',
      data: {
        __typename: 'UserNoteEntity',
        id: args.id,
      });
```
Is this what you meant ?
Also I'm not sure what is "detail resolution".

**Reply by JoviDeCroock** [reply]
List detail is documented [here](https://formidable.com/open-source/urql/docs/graphcache/local-resolvers/#resolving-entities)

And yes you are able to update links etc for the cache, this is explained [here](https://formidable.com/open-source/urql/docs/graphcache/cache-updates/#updating-lists-or-links)
